### PR TITLE
Grid improvements and fixes

### PR DESCRIPTION
**Pull request by aptkingston**
## Description
This PR contains further improvements to the grid based on feedback. It also addresses some unintended functionality where keyboard shortcuts could circumvent readonly cells.

Targeting master since I'm keen to get these out sooner rather than later, and some of these are indeed bug fixes rather than just improvements.

### Accessibility improvements for small screens
A significant improvement is the ability to work on much smaller screen sizes. The grid will now function on resolutions as small as 1280x720 (and even smaller):
image

To support this, some buttons collapse to icons which show tooltips on hover (on small screens only):
image

Row bulk deletion has been moved inline, instead of the button in the top right. Now when rows are selected, a delete icon will appear beside each row. Clicking any of these icons will prompt for bulk deletion:
image

### Auto column discoverability
2 changes have been made to support discovery of auto columns.

All auto columns now use the same icon in the grid header cells:
image

Icons have been added to the "hide columns" dropdown, showing the data type of each column. With the new dedicated icon for auto columns, this makes it much faster to tell which columns are auto columns:
image

### Other changes
- Fix issues with not honouring the delete rows flag
- Fix keyboard shortcuts being able to add rows when you shouldn't be able to
- Fix selecting rows with the keyboard when you shouldn't be able to
- Hide add column icon when you shouldn't be able to add columns
- Improve z-index of various elements to fix edge cases
- Move new row save and cancel buttons more left to avoid cases of overlapping dropdowns 
- Fix not being able to export selected rows, as selected rows were cleared when clicking the export button
- Improve other aspects of the "click outside" handler to fix issues with download-js, and improve click handling with modals

### CODE REVIEW

File: packages/frontend-core/src/components/grid/controls/HideColumnsButton.svelte [hunk 1]
```diff
@@ -47,25 +48,34 @@
...
+        <div class="column">
+          <Icon size="S" name={getColumnIcon($stickyColumn)} />
+          {$stickyColumn.label}
+        </div>
         <Toggle disabled size="S" value={true} />
-        <span>{$stickyColumn.label}</span>
       {/if}
       {#each $columns as column}
+        <div class="column">
+          <!--{#if column.schema.autocolumn}-->
```
Comment (shogunpurple) [inline packages/frontend-core/src/components/grid/controls/HideColumnsButton.svelte:69]: Can remove comment

**Review by shogunpurple — APPROVED** [review]
LGTM!

**Comment by mike12345567** [comment]
Also fixes: https://github.com/Budibase/budibase/issues/10490

**Comment by shogunpurple** [comment]
@aptkingston nice work on the quick fix. Is there any way for us to cover this one with a unit test for regression?

**Comment by aptkingston** [comment]
> @aptkingston nice work on the quick fix. Is there any way for us to cover this one with a unit test for regression?

Don't think it's unit testable - it would literally just be asserting that a svelte store works. I moved around how certain props were handled and this was just one reference I forgot to update (but I've searched and there aren't any more references to the old syntax now).

It would probably fit the criteria for a QA wolf integration test though, to verify all the basic features of the grid (and buttons above it).

**Review by mike12345567 — APPROVED** [review]
LGTM!

**Pull request merged**